components/builder-depot/src/server.rs

```diff
@@ -868,6 +866,14 @@ fn upload_package(req: &mut Request) -> IronResult<Response> {
 
     let filename = depot.archive_path(&ident, &target_from_artifact);
```

Comment:
I think we can completely remove the archive_path function, and just use the temp file as the path.  In the new S3 world, we should not need to have a carefully crafted archive path.

Reply:
This will also simplify and tighten up some other stuff.  For example, we are holding a read lock on depot for quite a while - if we remove the archive_path and archive_parent functions, we can tighten up the lock usage to only the place where we check for the targets.